models/issue.go

```diff
@@ -1348,22 +1348,50 @@ func (opts *IssuesOptions) setupSessionNoLimit(sess *xorm.Session) {
 	}
 
 	if opts.User != nil {
-		sess.And(
-			issuePullAccessibleRepoCond("issue.repo_id", opts.User.ID, opts.Org, opts.Team, opts.IsPull.IsTrue()),
-		)
+		repoCond, err := issuePullAccessibleRepoCond("issue.repo_id", opts.User.ID, opts.Org, opts.Team, opts.IsPull.IsTrue())
+		if err != nil {
+			return err
+		}
+		sess.And(repoCond)
 	}
+	return nil
 }
 
 // issuePullAccessibleRepoCond userID must not be zero, this condition require join repository table
-func issuePullAccessibleRepoCond(repoIDstr string, userID int64, org *organization.Organization, team *organization.Team, isPull bool) builder.Cond {
+func issuePullAccessibleRepoCond(repoIDstr string, userID int64, org *organization.Organization, team *organization.Team, isPull bool) (builder.Cond, error) {
 	cond := builder.NewCond()
 	unitType := unit.TypeIssues
 	if isPull {
 		unitType = unit.TypePullRequests
 	}
 	if org != nil {
 		if team != nil {
-			cond = cond.And(teamUnitsRepoCond(repoIDstr, userID, org.ID, team.ID, unitType)) // special team member repos
+			// If the current user is a admin, it doesn't necesarly mean
+			// it has joined the team, but we still should return all repo's
+			// of that team.
+			isAdmin, err := org.IsOwnedBy(userID)
```

Comment:
This will begin a new db connection that I don't know whether it will affect outside of the function.

Reply:
I dont see another way of checking if this person is a admin of the organization. They should "act" as they are member of the team.

Reply:
Should we inline the SQL that is used?

Reply:
Yes

Reply:
Deadlocks are real ;)